### ISSUE TITLE
ClassCastException when deserializing relationship which is parametrized type

### ISSUE THREAD

**Issue by jimirocks** [toedter/spring-hateoas-jsonapi#25]
When testing desrialization of polymorphic relationships I found this bug.

For instance having in relationship class like this:
```java
class NotWorking<T> extends SomeInterface<T>
```

```
java.lang.ClassCastException: class sun.reflect.generics.reflectiveObjects.WildcardTypeImpl cannot be cast to class java.lang.Class (sun.reflect.generics.reflectiveObjects.WildcardTypeImpl and java.lang.Class are in module java.base of loader 'bootstrap')
```

The problem is this line:

https://github.com/toedter/spring-hateoas-jsonapi/blob/8d4a5351431e3e1ba13631792c00745e4c17262c/lib/src/main/java/com/toedter/spring/hateoas/jsonapi/JsonApiEntityModelDeserializer.java#L106

The solution probably is to first obtain `ObjectMapper` and use it's `constructType` method which already covers all the (or at least reasonable subset) of `Type` descendants

**Comment by toedter** [comment]
Thx a lot, will review asap and then merge. Related to your question of `type ` vs. `_type`. I decided to use Jackson itself to resolve everything related to Jackson annotations, rather than parsing and interpreting them myself. Since this happened before the JSON:API specific deserialization, Jackson expects that type is a real property. Maybe there is a better solution that works without this restriction.

**Comment by jimirocks** [comment]
Thanks for merging my fix, do you have any schedule for 0.15.0 release?

**Comment by toedter** [comment]
I am working on some little improvements. But I could do a new release soon, maybe next weekend.

**Comment by jimirocks** [comment]
@toedter Do you already have schedule for 0.15.0? There are also other fixes which are becoming urgent to us. Thanks in advance

**Comment by toedter** [comment]
Sorry, I did not feel well in the last few days. I will try to release it in the next few days. The snapshot is also available publicly, you just have to add https://oss.sonatype.org/content/repositories/snapshots/ as a repository to your Maven or Gradle configuration.

**Comment by toedter** [comment]
Just released 0.15.0
